bitcoin/src/blockdata/transaction.rs

```diff
@@ -391,6 +397,26 @@ impl TransactionExt for Transaction {
         self.input.len() == 1 && self.input[0].previous_output == OutPoint::COINBASE_PREVOUT
     }
 
+    fn coinbase_height(&self) -> Result<u64, Bip34Error> {
+        if !self.is_coinbase() {
+            return Err(Bip34Error::NotPresent);
```

Comment:
Shouldn't this be `NotCoinbase`?

FTR, it feels like we might want a newtype for coinbsase.